cmd2/cmd2.py

```diff
@@ -595,10 +591,17 @@ def complete_submenu(_self, text, line, begidx, endidx):
             try:
                 # copy over any shared attributes
                 self._copy_in_shared_attrs(_self)
+                submenu.allow_appended_space = _self.allow_appended_space
+                submenu.allow_closing_quote = _self.allow_closing_quote
+                submenu.display_matches = _self.display_matches
+
```

Comment:
Instead of copying these attributes back and forth, why not put these attributes in its own object and pass a reference down the submenus? If submenus modify these values it will be on the same instance the outer layer is using.